test/test_autograd.py

```diff
@@ -1260,6 +1260,13 @@ def backward(ctx, grad_b):
 
         TestFn.apply(b).sum().backward()
 
+    def test_first_grad_fn_access_in_no_grad_mode(self):
+        a = torch.tensor([1 + 1j], requires_grad=True).clone()
+        v = a.real
+        a.add_(1)
+        with torch.autograd.grad_mode.no_grad():
+            v.grad_fn
```

Comment:
Don't you need to try and use it for the error to happen?

Reply:
Actually refreshing grad_fn also tries to update the cpp hooks, which checks that the grad_fn passed in is not nullptr

Reply:
Ok!